src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -197,10 +198,14 @@ private static bool TryGetValue(IDictionary<string, ISymbol> dictionary, string
                     if (TryGetValue(parameterToExistingFieldMap, parameterName, out fieldName) ||
                         TryGetValue(parameterToNewFieldMap, parameterName, out fieldName))
                     {
+                        var fieldReference = factory.IdentifierName(fieldName);
+
+                        var qualifiedFieldReference = StringComparer.OrdinalIgnoreCase.Equals(parameterName, fieldName)
```

Comment:
We always generate this qualified and let the simplifier remove the qualification if the user's settings don't require it.